pkg/api/resource/quantity.go

```diff
@@ -190,7 +190,9 @@ func ParseQuantity(str string) (*Quantity, error) {
 	// of an amount.
 	// Arguably, this should be inf.RoundHalfUp (normal rounding), but
 	// that would have the side effect of rounding values < .5m to zero.
-	amount.Round(amount, 3, inf.RoundUp)
+	if v, ok := amount.Unscaled(); v != int64(0) && ok {
```

Comment:
I think you want ```|| !ok```-- you should do the rounding if the Unscaled value doesn't fit in an int64.